### PR TITLE
feat(twilight-pref-179): Add display setting for twilight calculation

### LINKED ISSUE #179: Timeline dawn / dusk indications use astronomical twilight
The SunCalc code can support civil, nautical and astronomical twilight, and this should probably be a configuration option.

### PR DESCRIPTION
**Pull request by kwilson-bits**
Added a display setting for which of three twilight calculation methods is used on the timeline.
I also made a change to the destroy_() function to test if the removeResize() function exists before calling it. This was changed so that I could also add unit tests for the new functionality. Without this change, the destroy_() function threw errors.

fix #179

### CODE REVIEW

File: src/os/config/displaysettings.js [hunk 1]
```diff
@@ -55,7 +55,8 @@ os.config.DisplaySetting = {
   ENABLE_SKY: os.config.DisplaySettings.BASE_KEY + 'enableSky',
   ENABLE_LIGHTING: os.config.DisplaySettings.BASE_KEY + 'enableLighting',
   ENABLE_TERRAIN: os.config.DisplaySettings.BASE_KEY + 'enableTerrain',
-  TERRAIN_OPTIONS: os.config.DisplaySettings.BASE_KEY + 'terrainOptions'
+  TERRAIN_OPTIONS: os.config.DisplaySettings.BASE_KEY + 'terrainOptions',
+  DUSK_MODE: os.config.DisplaySettings.BASE_KEY + 'duskMode'
```
Comment (wallw-bits) [inline src/os/config/displaysettings.js]: This is a plugin feature (`plugin.x`) that is being added and referenced in a core class (`os.x`). It is better and more modular to have the plugin create and register a settings UI.

File: src/os/map/map.js [hunk 2]
```diff
@@ -49,6 +49,17 @@ os.CameraMode = {
...
 
+/**
+ * Modes for displaying dusk duration in the timeline.
+ * @enum {string}
+ */
+os.DuskMode = {
+  CIVILIAN: 'civilian',
+  NAUTICAL: 'nautical',
+  ASTRONOMICAL: 'astronomical'
+};
```
Comment (wallw-bits) [inline src/os/map/map.js]: Same here.

File: src/plugin/suncalc/lightstrip.js [hunk 3]
```diff
@@ -84,7 +101,9 @@ plugin.suncalc.LightStripCtrl.prototype.destroy_ = function() {
   this.updateDelay_ = null;
 
   if (this.element_ && this.resizeFn_) {
-    this.element_.parent().removeResize(this.resizeFn_);
+    if (typeof this.element_.parent().removeResize === 'function') {
```
Comment (schmidtk) [inline src/plugin/suncalc/lightstrip.js]: Please remove this, and instead add `jquery.resize.js` to the Karma configuration.

File: src/plugin/suncalc/lightstripsettings.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+plugin.suncalc.LightStripSettingsCtrl = function($scope) {
+  /**
+   * @type {?angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  this.scope_.$on('$destroy', this.destroy_.bind(this));
+
+  this['twilight_calculation'] = /** @type {string} */ (os.settings.get(plugin.suncalc.SettingKey.DUSK_MODE,
```
Comment (schmidtk) [inline src/plugin/suncalc/lightstripsettings.js]: Minor, but use camel case (`twilightCalculation`) for properties.

**Pull request merged**